### ISSUE TITLE
[Question] test.setTimeout() in test hooks

### ISSUE THREAD

**Issue by seunghunee** [microsoft/playwright#7266]
I have an expensive calculation in `beforeAll`.
So, I need to increase the timeout value for this specific test. (playwright default `30000`)
When I used Jest, I just put `jest.setTimeout()` top of the file.

Now, I'm using Playwright Test. And I found `test.setTimeout()` which looks similar to `jest.setTimeout()`.
But, `test.setTimeout()` can only be called inside 1.test or 2.fixture.
I tried 1. but failed because heavy calculation was in `beforeAll` hook.

Do I have to use fixture? How to solve this problem?

**Comment by dgozman** [comment]
@seunghunee I think we can do a top-level (or describe-level) setTimeout:

```js
test.beforeAll(async () => {
  // Long beforeAll hook...
});

// Increase timeout for all tests in this file.
test.setTimeout(120000);

test('my test', async ({ page }) => {
});
```

Does that work for you?

**Comment by seunghunee** [comment]
I've tried once again on v1.12.3 for both cases (top-level and describe-level)
But, it doesn't work. I got this error message
```
test.setTimeout() can only be called inside test or fixture
```

**Comment by dgozman** [comment]
@seunghunee Sorry for not being clear. This does not work in v1.12.3, I was asking whether this API will satisfy your needs.

**Comment by dgozman** [comment]
The fix will be available in v1.13.0

**Comment by seunghunee** [comment]
@dgozman Thank you so much!
